### PR TITLE
concord: init at 2.2.0

### PR DESCRIPTION
**Pull request by transgirllucy**
###### Description of changes

I made it compatible with Nix

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/concord/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "v${version}";
+    hash = "sha256-+nuVyqA/V/J6DWAeIs9Pv90ry3px1gJsF460qvfSEH8="; # Fill this in based on the fetch error
+  };
+
+  buildInputs = [
+    curl
+  ];
+
+  installFlags = [
+  	"PREFIX=${"out"}"
```
Comment (NobbZ) [inline pkgs/development/libraries/concord/default.nix:23]: IIRC this is basically the default. I think you can safely remove the `installFlags`

**Review by NobbZ — CHANGES_REQUESTED** [review]
1. Please also add a commit *before* the commit adding the package, that adds yourself to the maintainers set.
2. Please add the package to the all-packages list
3. Use spaces, not tabs: https://github.com/NixOS/nixpkgs/actions/runs/4045587446/jobs/6957254479#step:7:8

File: pkgs/development/libraries/concord/default.nix [hunk 2]
```diff
@@ -1,8 +1,4 @@
-{ stdenv
-, lib
-, fetchFromGitHub
-, curl
-}:
+{ stdenv, lib, fetchFromGitHub, curl }:
```
Comment (NobbZ) [inline pkgs/development/libraries/concord/default.nix:1]: Please keep them on seperate lines, makes future additions/removals/changes easier to diff

File: pkgs/development/libraries/concord/default.nix [hunk 3]
```diff
@@ -12,16 +8,13 @@ stdenv.mkDerivation rec {
     owner = "Cogmasters";
     repo = "concord";
     rev = "v${version}";
-    hash = "sha256-+nuVyqA/V/J6DWAeIs9Pv90ry3px1gJsF460qvfSEH8="; # Fill this in based on the fetch error
+    hash =
+      "sha256-+nuVyqA/V/J6DWAeIs9Pv90ry3px1gJsF460qvfSEH8=";
```
Comment (NobbZ) [inline pkgs/development/libraries/concord/default.nix:12]: Please keep this on a single line.

File: pkgs/development/libraries/concord/default.nix [hunk 4]
```diff
@@ -12,16 +8,13 @@ stdenv.mkDerivation rec {
...
   };
 
-  buildInputs = [
-    curl
-  ];
+  buildInputs = [ curl ];
```
Comment (NobbZ) [inline pkgs/development/libraries/concord/default.nix:15]: Please keep them on seperate lines, makes future additions/removals/changes easier to diff

File: pkgs/development/libraries/concord/default.nix [hunk 5]
```diff
@@ -12,16 +8,13 @@ stdenv.mkDerivation rec {
...
 
-  installFlags = [
-  	"PREFIX=${"out"}"
-  ];
+  installFlags = [ "PREFIX=${"out"}" ];
```
Comment (NobbZ) [inline pkgs/development/libraries/concord/default.nix:17]: As said earlier, it should be save to remove the `installFlags`.

**Review by NobbZ — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/concord/default.nix (hunk 1 above)
Comment (NobbZ) [reply pkgs/development/libraries/concord/default.nix]: Then it was the default for configuration. Which is not done here. Okay. Will resolve.

File: pkgs/development/libraries/concord/default.nix (hunk 5 above)
Comment (NobbZ) [reply pkgs/development/libraries/concord/default.nix:17]: Okay, needs to stay, *but* please use seperate lines, diff and future changes…

**Review by NobbZ — CHANGES_REQUESTED** [review]
1. Remove the trailing whitespace introduced by the last commit
2. read the other open change requests, especially adding appropriate entries to the `maintainers/maintainer-list.nix` and `pkgs/top-level/all-packages.nix` are crucial to be even able to access this expression!

File: pkgs/development/libraries/concord/default.nix (hunk 2 above)
Comment (NobbZ) [reply pkgs/development/libraries/concord/default.nix:1]: 👆

**Comment by NobbZ** [comment]
Please remove the trailing spaces, as requested by the failing check:

https://github.com/NixOS/nixpkgs/actions/runs/4059397569/jobs/6987393496

Also, please squash and reorder your commits such that there exist exactly 2 and in this order:

1. Add yourself to the maintainers list with the subject "maintainers: Add mrtuxa as a maintainer"
2. Create the package and the entry in `all-packages` with the subject "concord: init at 2.2.0"

**Comment by mrtuxa** [comment]
> Please remove the trailing spaces, as requested by the failing check:
> 
> https://github.com/NixOS/nixpkgs/actions/runs/4059397569/jobs/6987393496
> 
> Also, please squash and reorder your commits such that there exist exactly 2 and in this order:
> 
> 1. Add yourself to the maintainers list with the subject "maintainers: Add mrtuxa as a maintainer"
> 2. Create the package and the entry in `all-packages` with the subject "concord: init at 2.2.0"

i already created the project how can i add the subject

**Comment by mrtuxa** [comment]
i only need to squash everything?

**Comment by NobbZ** [comment]
> i only need to squash everything?

No, you need to redo the commits such that the 2 commits are created as explained above.

One for the mainters list and one for the package. In that order.

**Comment by mrtuxa** [comment]
> > i only need to squash everything?
> 
> No, you need to redo the commits such that the 2 commits are created as explained above.
> 
> One for the mainters list and one for the package. In that order.

how can i do that, sound quiet difficult

**Comment by NobbZ** [comment]
Actually rebases are quite easy. The problem you have here is, that you already have been sloppy with your commits and that you have a commit that already touches maintainers file and something else.

You now have to split that commit up into 2.

It might be the easiest to do a rebase in which you move 15c085b078cd45fb2436369b23d5cc03695d044c to the first position, edit it to not touch anything but the maintainers list, and have all the other commits thereafter, squash/fixup them into a single commit for the package related stuff.

Go through all the changes again, take a carefull look if all the suggested changes have been applied.

If necessary amend the corresponding commit.

After all changes have been applied, force push.

----

If this is too much for you, create a new PR being made more carefully following the change requests given here, and having everything in the 2 commits as explained.

Be warned though, that any change you are asked for then, still has to be squashed into one of the 2 commits anyway.

You will need to learn how to git.

**Comment by mrtuxa** [comment]
i will create a new PR

**Pull request closed without merging**